sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -380,9 +379,24 @@ impl<T: Flavor> Transport for Hyperdebug<T> {
 
     // Crate SPI Target instance, or return one from a cache of previously created instances.
     fn spi(&self, instance: &str) -> Result<Rc<dyn Target>> {
-        let &idx = self.spi_names.get(instance).ok_or_else(|| {
-            TransportError::InvalidInstance(TransportInterfaceType::Spi, instance.to_string())
-        })?;
+        // Execute a "spiget" command to look up the numeric index corresponding to the given
+        // textual SPI instance name.
+        let mut result: Result<u8> = Err(TransportError::InvalidInstance(
+            TransportInterfaceType::Spi,
+            instance.to_string(),
+        )
+        .into());
+        let regex = Regex::new("^ +([0-9]+) ([^ ]+) ([0-9]+)")?;
```

Comment:
Should this be a created once at program init time?
```
lazy_static! {
  static ref REGEX: Regex = Regex::new(...).unwrap();
}

Reply:
Done.